webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/volume/VolumeTracingService.scala

```diff
@@ -343,36 +213,20 @@ class VolumeTracingService @Inject()(
     val dataLayer = volumeTracingLayer(tracingId, tracing)
     val savedResolutions = new mutable.HashSet[Point3D]()
 
-    val unzipResult = ZipIO.withUnziped(initialData) {
-      case (fileName, is) =>
-        WKWFile.read(is) {
-          case (header, buckets) =>
-            if (header.numBlocksPerCube == 1) {
-              parseWKWFilePath(fileName.toString).map { bucket =>
-                if (buckets.hasNext) {
-                  val data = buckets.next()
-                  if (isAllZero(data) || resolutionRestrictions.isForbidden(bucket.resolution)) {
-                    Fox.successful(())
-                  } else {
-                    savedResolutions.add(bucket.resolution)
-                    saveBucket(dataLayer, bucket, data, tracing.version)
-                  }
-                }
-              }
-            }
-        }
+    val unzipResult = withBucketsFromZip(initialData) { (bucketPosition, bytes) =>
+      if (resolutionRestrictions.isForbidden(bucketPosition.resolution)) {
+        Fox.successful(())
+      } else {
+        savedResolutions.add(bucketPosition.resolution)
+        saveBucket(dataLayer, bucketPosition, bytes, tracing.version)
+      }
     }
 
     for {
       _ <- unzipResult
     } yield savedResolutions.toSet
```

Comment:
I would prefer a simple  `.map()` when only having one statement in a `for ... yield`, but that's personal preference I guess.

Maybe we should discuss in person, also for the future

Reply:
I agree, map is better. I had originally written `unzipResult.map(savedResolutions.toSet)` (forgetting the `_ =>`) and had changed it to the for comprehension when that didn’t compile. I changed it to the correct map format now :)